gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -125,81 +181,127 @@ protected void createMetrics() {
   protected List<Tag<?>> getTagsForMetrics() {
     List<Tag<?>> tags = Lists.newArrayList();
     tags.add(new Tag<>(RuntimeMetrics.TOPIC, this.topic));
-    tags.add(new Tag<>(RuntimeMetrics.GROUP_ID, this.consumerConfig.groupId()));
+    tags.add(new Tag<>(RuntimeMetrics.GROUP_ID, ConfigUtils.getString(this.config, GROUP_ID_KEY, DEFAULT_GROUP_ID)));
     return tags;
   }
 
   /**
-   * Called every time a message is read from the stream. Implementation must be thread-safe if {@link #numThreads} is
+   * Called every time a message is read from the queue. Implementation must be thread-safe if {@link #numThreads} is
    * set larger than 1.
    */
-  protected abstract void processMessage(MessageAndMetadata<K, V> message);
+  protected abstract void processMessage(DecodeableKafkaRecord<K,V> message);
 
   @Override
   protected void startUp() {
     buildMetricsContextAndMetrics();
-    this.consumer = createConsumerConnector();
+    processQueues();
+    if(!enableAutoCommit) {
+      offsetCommitExecutor = Executors.newSingleThreadScheduledExecutor();
+      offsetCommitExecutor.scheduleAtFixedRate(new Runnable() {
+        @Override
+        public void run() {
+          offsetCommitter();
```

Comment:
offsetCommitter() -> commitOffsets()? Prefer method names to always be actions.
https://howtodoinjava.com/java/basics/java-naming-conventions/